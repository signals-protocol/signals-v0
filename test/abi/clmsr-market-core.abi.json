[
  "error AddressEmptyCode(address)",
  "error AffectedSumZero()",
  "error ArrayLengthMismatch()",
  "error BinCountExceedsLimit(uint32,uint32)",
  "error BinOutOfBounds(uint32,uint32)",
  "error ChunkLimitExceeded(uint256,uint256)",
  "error CostExceedsMaximum(uint256,uint256)",
  "error ERC1967InvalidImplementation(address)",
  "error ERC1967NonPayable()",
  "error EnforcedPause()",
  "error ExpectedPause()",
  "error FactorOutOfBounds()",
  "error FailedCall()",
  "error IncompleteChunkProcessing()",
  "error IndexOutOfBounds(uint32,uint32)",
  "error InsufficientBalance(address,uint256,uint256)",
  "error InsufficientPositionQuantity(uint128,uint128)",
  "error InvalidFactor(uint256)",
  "error InvalidInitialization()",
  "error InvalidLiquidityParameter()",
  "error InvalidMarketParameters(int256,int256,int256)",
  "error InvalidQuantity(uint128)",
  "error InvalidRange(uint32,uint32)",
  "error InvalidRangeBins(uint32,uint32)",
  "error InvalidRangeCount(int256,uint256)",
  "error InvalidTick(int256,int256,int256)",
  "error InvalidTickRange(int256,int256)",
  "error InvalidTickSpacing(int256,int256)",
  "error InvalidTimeRange()",
  "error InvalidTokenDecimals(uint8,uint8)",
  "error LazyFactorOverflow()",
  "error ManagerNotSet()",
  "error MarketAlreadyExists(uint256)",
  "error MarketAlreadySettled(uint256)",
  "error MarketExpired()",
  "error MarketNotActive()",
  "error MarketNotFound(uint256)",
  "error MarketNotSettled(uint256)",
  "error MarketNotStarted()",
  "error MathMulOverflow()",
  "error NoChunkProgress()",
  "error NonIncreasingSum(uint256,uint256)",
  "error NotInitializing()",
  "error OwnableInvalidOwner(address)",
  "error OwnableUnauthorizedAccount(address)",
  "error PositionNotFound(uint256)",
  "error ProceedsBelowMinimum(uint256,uint256)",
  "error QuantityOverflow()",
  "error RangeBinOutOfBounds(int256,uint32)",
  "error RangeBinsOutOfBounds(uint32,uint32,uint32)",
  "error ReentrancyGuardReentrantCall()",
  "error ResidualQuantity(uint256)",
  "error SafeERC20FailedOperation(address)",
  "error SettlementTooEarly(uint64,uint64)",
  "error SumAfterZero()",
  "error TreeAlreadyInitialized()",
  "error TreeNotInitialized()",
  "error TreeSizeTooLarge()",
  "error TreeSizeZero()",
  "error UUPSUnauthorizedCallContext()",
  "error UUPSUnsupportedProxiableUUID(bytes32)",
  "error UnauthorizedCaller(address)",
  "error ZeroAddress()",
  "error ZeroLimit()",
  "event EmergencyPaused(indexed address,string)",
  "event EmergencyUnpaused(indexed address)",
  "event Initialized(uint64)",
  "event ManagerUpdated(indexed address,indexed address)",
  "event MarketActivationUpdated(indexed uint256,bool)",
  "event MarketCreated(indexed uint256,uint64,uint64,int256,int256,int256,uint32,uint256)",
  "event MarketReopened(indexed uint256)",
  "event MarketSettled(indexed uint256,int256)",
  "event MarketSettlementValueSubmitted(indexed uint256,int256)",
  "event MarketTimingUpdated(indexed uint256,uint64,uint64)",
  "event OwnershipTransferred(indexed address,indexed address)",
  "event Paused(address)",
  "event PositionClaimed(indexed uint256,indexed address,uint256)",
  "event PositionClosed(indexed uint256,indexed address,uint256)",
  "event PositionDecreased(indexed uint256,indexed address,uint128,uint128,uint256)",
  "event PositionEventsProgress(indexed uint256,uint256,uint256,bool)",
  "event PositionIncreased(indexed uint256,indexed address,uint128,uint128,uint256)",
  "event PositionOpened(indexed uint256,indexed address,indexed uint256,int256,int256,uint128,uint256)",
  "event PositionSettled(indexed uint256,indexed address,uint256,bool)",
  "event RangeFactorApplied(indexed uint256,indexed int256,indexed int256,uint256)",
  "event RangeFactorBatchApplied(indexed uint256,uint256,bytes32)",
  "event SettlementTimestampUpdated(indexed uint256,uint64)",
  "event Unpaused(address)",
  "event Upgraded(indexed address)",
  "function MAX_LIQUIDITY_PARAMETER view() returns (uint256)",
  "function MAX_TICK_COUNT view() returns (uint32)",
  "function MIN_LIQUIDITY_PARAMETER view() returns (uint256)",
  "function UPGRADE_INTERFACE_VERSION view() returns (string)",
  "function _nextMarketId view() returns (uint256)",
  "function applyRangeFactor(uint256,int256,int256,uint256)",
  "function applyRangeFactorBatch(uint256,int256[],int256[],uint256[],bytes32)",
  "function calculateClaimAmount view(uint256) returns (uint256)",
  "function calculateCloseProceeds view(uint256) returns (uint256)",
  "function calculateDecreaseProceeds view(uint256,uint128) returns (uint256)",
  "function calculateIncreaseCost view(uint256,uint128) returns (uint256)",
  "function calculateOpenCost view(uint256,int256,int256,uint128) returns (uint256)",
  "function calculateQuantityFromCost view(uint256,int256,int256,uint256) returns (uint128)",
  "function claimPayout(uint256) returns (uint256)",
  "function closePosition(uint256,uint256) returns (uint256)",
  "function createMarket(int256,int256,int256,uint64,uint64,uint64,uint256) returns (uint256)",
  "function decreasePosition(uint256,uint128,uint256) returns (uint128,uint256)",
  "function emitPositionSettledBatch(uint256,uint256)",
  "function getMarket view(uint256) returns ((bool,bool,uint64,uint64,int256,int256,int256,int256,uint32,uint256,uint32,bool,int256,uint64))",
  "function getPaymentToken view() returns (address)",
  "function getPositionContract view() returns (address)",
  "function getRangeSum view(uint256,int256,int256) returns (uint256)",
  "function increasePosition(uint256,uint128,uint256) returns (uint128)",
  "function initialize(address,address)",
  "function isPaused view() returns (bool)",
  "function manager view() returns (address)",
  "function marketTrees view(uint256) returns (uint32,uint32,uint32,uint256)",
  "function markets view(uint256) returns (bool,bool,uint64,uint64,int256,int256,int256,int256,uint32,uint256,uint32,bool,int256,uint64)",
  "function openPosition(uint256,int256,int256,uint128,uint256) returns (uint256)",
  "function owner view() returns (address)",
  "function pause(string)",
  "function paused view() returns (bool)",
  "function paymentToken view() returns (address)",
  "function positionContract view() returns (address)",
  "function positionSettledEmitted view(uint256) returns (bool)",
  "function propagateLazy(uint256,int256,int256) returns (uint256)",
  "function proxiableUUID view() returns (bytes32)",
  "function renounceOwnership()",
  "function reopenMarket(uint256)",
  "function setManager(address)",
  "function setMarketActive(uint256,bool)",
  "function settleMarket(uint256,int256)",
  "function transferOwnership(address)",
  "function unpause()",
  "function updateMarketTiming(uint256,uint64,uint64,uint64)",
  "function upgradeToAndCall payable(address,bytes)"
]
