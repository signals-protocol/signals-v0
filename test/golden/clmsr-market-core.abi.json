[
  "error AddressEmptyCode(address)",
  "error AffectedSumZero()",
  "error BinCountExceedsLimit(uint32,uint32)",
  "error ChunkLimitExceeded(uint256,uint256)",
  "error CostExceedsMaximum(uint256,uint256)",
  "error ERC1967InvalidImplementation(address)",
  "error ERC1967NonPayable()",
  "error EnforcedPause()",
  "error ExpectedPause()",
  "error FactorOutOfBounds()",
  "error FailedCall()",
  "error IncompleteChunkProcessing()",
  "error InsufficientBalance(address,uint256,uint256)",
  "error InsufficientPositionQuantity(uint128,uint128)",
  "error InvalidInitialization()",
  "error InvalidLiquidityParameter()",
  "error InvalidMarketParameters(int256,int256,int256)",
  "error InvalidQuantity(uint128)",
  "error InvalidRangeBins(uint32,uint32)",
  "error InvalidRangeCount(int256,uint256)",
  "error InvalidTick(int256,int256,int256)",
  "error InvalidTickRange(int256,int256)",
  "error InvalidTickSpacing(int256,int256)",
  "error InvalidTimeRange()",
  "error InvalidTokenDecimals(uint8,uint8)",
  "error MarketAlreadyExists(uint256)",
  "error MarketAlreadySettled(uint256)",
  "error MarketExpired()",
  "error MarketNotActive()",
  "error MarketNotFound(uint256)",
  "error MarketNotSettled(uint256)",
  "error MarketNotStarted()",
  "error MathMulOverflow()",
  "error NoChunkProgress()",
  "error NonIncreasingSum(uint256,uint256)",
  "error NotInitializing()",
  "error OwnableInvalidOwner(address)",
  "error OwnableUnauthorizedAccount(address)",
  "error ProceedsBelowMinimum(uint256,uint256)",
  "error QuantityOverflow()",
  "error RangeBinsOutOfBounds(uint32,uint32,uint32)",
  "error ReentrancyGuardReentrantCall()",
  "error ResidualQuantity(uint256)",
  "error SafeERC20FailedOperation(address)",
  "error SettlementTooEarly(uint64,uint64)",
  "error SumAfterZero()",
  "error TreeNotInitialized()",
  "error UUPSUnauthorizedCallContext()",
  "error UUPSUnsupportedProxiableUUID(bytes32)",
  "error UnauthorizedCaller(address)",
  "error ZeroAddress()",
  "error ZeroLimit()",
  "event DebugSellProceeds(uint256,uint256,uint256,string)",
  "event EmergencyPaused(indexed address,string)",
  "event EmergencyUnpaused(indexed address)",
  "event Initialized(uint64)",
  "event MarketCreated(indexed uint256,uint64,uint64,int256,int256,int256,uint32,uint256)",
  "event MarketReopened(indexed uint256)",
  "event MarketSettled(indexed uint256,int256)",
  "event MarketSettlementValueSubmitted(indexed uint256,int256)",
  "event MarketTimingUpdated(indexed uint256,uint64,uint64)",
  "event OwnershipTransferred(indexed address,indexed address)",
  "event Paused(address)",
  "event PositionClaimed(indexed uint256,indexed address,uint256)",
  "event PositionClosed(indexed uint256,indexed address,uint256)",
  "event PositionDecreased(indexed uint256,indexed address,uint128,uint128,uint256)",
  "event PositionEventsProgress(indexed uint256,uint256,uint256,bool)",
  "event PositionIncreased(indexed uint256,indexed address,uint128,uint128,uint256)",
  "event PositionOpened(indexed uint256,indexed address,indexed uint256,int256,int256,uint128,uint256)",
  "event PositionSettled(indexed uint256,indexed address,uint256,bool)",
  "event RangeFactorApplied(indexed uint256,indexed int256,indexed int256,uint256)",
  "event SettlementTimestampUpdated(indexed uint256,uint64)",
  "event Unpaused(address)",
  "event Upgraded(indexed address)",
  "function MAX_LIQUIDITY_PARAMETER view() returns (uint256)",
  "function MAX_TICK_COUNT view() returns (uint32)",
  "function MIN_LIQUIDITY_PARAMETER view() returns (uint256)",
  "function UPGRADE_INTERFACE_VERSION view() returns (string)",
  "function _nextMarketId view() returns (uint256)",
  "function applyRangeFactor(uint256,int256,int256,uint256)",
  "function calculateClaimAmount view(uint256) returns (uint256)",
  "function calculateCloseProceeds view(uint256) returns (uint256)",
  "function calculateDecreaseProceeds view(uint256,uint128) returns (uint256)",
  "function calculateIncreaseCost view(uint256,uint128) returns (uint256)",
  "function calculateOpenCost view(uint256,int256,int256,uint128) returns (uint256)",
  "function calculateQuantityFromCost view(uint256,int256,int256,uint256) returns (uint128)",
  "function claimPayout(uint256) returns (uint256)",
  "function closePosition(uint256,uint256) returns (uint256)",
  "function createMarket(int256,int256,int256,uint64,uint64,uint64,uint256) returns (uint256)",
  "function decreasePosition(uint256,uint128,uint256) returns (uint128,uint256)",
  "function emitPositionSettledBatch(uint256,uint256)",
  "function getMarket view(uint256) returns ((bool,bool,uint64,uint64,int256,int256,int256,int256,uint32,uint256,uint32,bool,int256,uint64))",
  "function getPaymentToken view() returns (address)",
  "function getPositionContract view() returns (address)",
  "function getRangeSum view(uint256,int256,int256) returns (uint256)",
  "function increasePosition(uint256,uint128,uint256) returns (uint128)",
  "function initialize(address,address)",
  "function isPaused view() returns (bool)",
  "function marketTrees view(uint256) returns (uint32,uint32,uint32,uint256)",
  "function markets view(uint256) returns (bool,bool,uint64,uint64,int256,int256,int256,int256,uint32,uint256,uint32,bool,int256,uint64)",
  "function openPosition(uint256,int256,int256,uint128,uint256) returns (uint256)",
  "function owner view() returns (address)",
  "function pause(string)",
  "function paused view() returns (bool)",
  "function paymentToken view() returns (address)",
  "function positionContract view() returns (address)",
  "function positionSettledEmitted view(uint256) returns (bool)",
  "function propagateLazy(uint256,int256,int256) returns (uint256)",
  "function proxiableUUID view() returns (bytes32)",
  "function renounceOwnership()",
  "function reopenMarket(uint256)",
  "function settleMarket(uint256,int256)",
  "function transferOwnership(address)",
  "function unpause()",
  "function updateMarketTiming(uint256,uint64,uint64,uint64)",
  "function upgradeToAndCall payable(address,bytes)"
]
